cpp/cmake-modules/substrait.cmake

```diff
@@ -0,0 +1,68 @@
+# Copyright(c) 2022-2023 Intel Corporation.
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+macro(build_substrait)
+  message(STATUS "Generating substrait files")
+
+  if(NOT DEFINED SUBSTRAIT_ROOT)
+    message(FATAL_ERROR "SUBSTRAIT_ROOT not defined, please set SUBSTRAIT_ROOT")
+  endif()
+  message(STATUS "SUBSTRAIT_ROOT: ${SUBSTRAIT_ROOT}")
+
+  # Set up Proto
+  find_package(Protobuf REQUIRED)
+  find_package(FMT REQUIRED)
+
+  set(PROTO_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/thirdparty/substrait")
+  set(SUBSTRAIT_INCLUDE_DIR ${PROTO_OUTPUT_DIR})
+  set(substrait_proto_directory ${SUBSTRAIT_ROOT}/substrait)
+  file(MAKE_DIRECTORY ${PROTO_OUTPUT_DIR})
+  file(GLOB PROTO_FILES ${substrait_proto_directory}/*.proto
+       ${substrait_proto_directory}/extensions/*.proto)
+  foreach(PROTO ${PROTO_FILES})
+    file(RELATIVE_PATH REL_PROTO ${substrait_proto_directory} ${PROTO})
+    string(REGEX REPLACE "\\.proto" "" PROTO_NAME ${REL_PROTO})
```

Comment:
seems we can delete those sed fixes in Makefile.

Reply:
can't, these sed fixes are for velox build